zerver/lib/actions.py

```diff
@@ -3639,21 +3639,51 @@ def do_change_stream_invite_only(stream: Stream, invite_only: bool,
     )
     stream.invite_only = invite_only
     stream.history_public_to_subscribers = history_public_to_subscribers
-    stream.save(update_fields=['invite_only', 'history_public_to_subscribers'])
+    stream.is_web_public = False
+    stream.save(update_fields=['invite_only', 'history_public_to_subscribers', 'is_web_public'])
     event = dict(
         op="update",
         type="stream",
         property="invite_only",
         value=invite_only,
         history_public_to_subscribers=history_public_to_subscribers,
+        is_web_public=False,
         stream_id=stream.id,
         name=stream.name,
     )
     send_event(stream.realm, event, can_access_stream_user_ids(stream))
 
-def do_change_stream_web_public(stream: Stream, is_web_public: bool) -> None:
-    stream.is_web_public = is_web_public
-    stream.save(update_fields=['is_web_public'])
+def do_make_stream_web_public(stream: Stream) -> None:
+    stream.is_web_public = True
+    stream.invite_only = False
+    stream.history_public_to_subscribers = True
+    stream.save(update_fields=['invite_only', 'history_public_to_subscribers', 'is_web_public'])
+
+    # We reuse "invite_only" stream update API route here because
+    # both are similar events and similar UI updates will be required
+    # by the client to update this property for the user.
+    event = dict(
+        op="update",
+        type="stream",
+        property="invite_only",
+        value=False,
+        history_public_to_subscribers=True,
+        is_web_public=True,
+        stream_id=stream.id,
+        name=stream.name,
+    )
+    send_event(stream.realm, event, can_access_stream_user_ids(stream))
```

Comment:
@timabbott Should we send this event to all users?, since web public stream is accessible by everybody.